vllm/v1/core/sched/scheduler.py

```diff
@@ -1248,19 +1262,30 @@ def make_stats(
         spec_decoding_stats: SpecDecodingStats | None = None,
         kv_connector_stats: KVConnectorStats | None = None,
     ) -> SchedulerStats | None:
-        if not self.log_stats:
+        if not (self.log_stats or self._kv_metrics_enabled):
```

Comment:
KV metrics should only be enabled if log stats is also enabled ... `--disable-log-stats` turns off all metric collection and logging. I don't think we want to allow for a case where only KV metrics is enabled

i.e. please undo most of the changes to this method, and just add the KV metrics support